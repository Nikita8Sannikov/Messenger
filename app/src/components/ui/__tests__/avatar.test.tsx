import { render, screen } from "@testing-library/react";

import AvatarComponent from "@components/ui/avatar";

describe("AvatarComponent", () => {
	const mockProps = {
		src: "https://example.com/avatar.jpg",
		alt: "Test Avatar",
		userId: "123",
		id: "test-avatar",
		className: "custom-class",
	};

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("should render with avatar image", () => {
		render(<AvatarComponent {...mockProps} />);
        
		const avatar = screen.getByRole("img");
		expect(avatar).toBeInTheDocument();
		expect(avatar).toHaveAttribute("src", "https://example.com/avatar.jpg");
		expect(avatar).toHaveAttribute("alt", mockProps.alt);
	});

	it("should render with custom className", () => {
		render(<AvatarComponent {...mockProps} />);
        
		const avatar = screen.getByRole("img").parentElement;
		expect(avatar).toHaveClass("custom-class");
	});

	it("should render with custom id", () => {
		render(<AvatarComponent {...mockProps} />);
        
		const avatar = screen.getByRole("img").parentElement;
		expect(avatar).toHaveAttribute("id", mockProps.id);
	});

	it("should render with null src", () => {
		render(
			<AvatarComponent 
				{...mockProps} 
				src={null}
			/>,
		);
        
		const avatar = screen.getByRole("img");
		expect(avatar).toHaveAttribute("src", "/assets/images/noPhoto.jpg");
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<AvatarComponent {...mockProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
