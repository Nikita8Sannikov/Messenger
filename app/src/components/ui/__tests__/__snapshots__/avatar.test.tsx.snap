// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarComponent matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiBadge-root avatar-badge css-63ezxx-MuiBadge-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular avatar custom-class css-fmeqll-MuiAvatar-root"
      id="test-avatar"
    >
      <img
        alt="Test Avatar"
        class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
        src="https://example.com/avatar.jpg"
      />
    </div>
    <span
      class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-aa1yeu-MuiBadge-badge"
    />
  </span>
</DocumentFragment>
`;
